# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/Document.cpp
# Commit: 72e19189bb99
# Full Hash: 72e19189bb99e159b2974cbcda69647211962df3
# Author: Brendan Dahl <bdahl@mozilla.com>
# Date: 2019-06-14 03:39:39
# Regressor Bug: 1552049
# File Overlap Count: 1
# Description:
#   Bug 1552049 - Enable XUL persistence for all system privilege docs. r=smaug
#   
#   Remove the mozpersist attribute and XULDocument requirement to use XUL
#   persistence and allow any system privilege document to use it by default.
#   
# ==============================================================================

diff -r bc46a287b898 -r 72e19189bb99 dom/base/Document.cpp
--- a/dom/base/Document.cpp	Thu Jun 13 17:37:15 2019 +0000
+++ b/dom/base/Document.cpp	Thu Jun 13 18:15:54 2019 +0000
@@ -11023,12 +11023,8 @@
 
   if (READYSTATE_INTERACTIVE == rs) {
     if (nsContentUtils::IsSystemPrincipal(NodePrincipal())) {
-      Element* root = GetRootElement();
-      if ((root && root->HasAttr(kNameSpaceID_None, nsGkAtoms::mozpersist)) ||
-          IsXULDocument()) {
-        mXULPersist = new XULPersist(this);
-        mXULPersist->Init();
-      }
+      mXULPersist = new XULPersist(this);
+      mXULPersist->Init();
     }
   }
 